crates/environ/src/stack_switching.rs

```diff
@@ -0,0 +1,78 @@
+//! This module contains basic type definitions used by the implementation of
+//! the stack switching proposal.
+
+/// FIXME(frank-emrich) Will remove in the final upstreamed version
+#[allow(dead_code, reason = "Only accessed in debug builds")]
+pub const ENABLE_DEBUG_PRINTING: bool = false;
+
+/// FIXME(frank-emrich) Will remove in the final upstreamed version
+#[macro_export]
+macro_rules! debug_println {
+    ($( $args:expr ),+ ) => {
+        #[cfg(debug_assertions)]
+        if ENABLE_DEBUG_PRINTING {
+            #[cfg(feature = "std")]
+            println!($($args),*);
+        }
+    }
+}
```

Comment:
fyi, I tend to do `log::trace!(...)` for debug prints (both temporary ones that I will remove, and for ones that are useful enough to keep in tree).

when running the CLI, you can see them by setting the `WASMTIME_LOG=trace` env var, or only a specific crate/module via `WASMTIME_LOG=wasmtime::runtime::vm=trace`, or multiple crates/modules at different levels via `WASMTIME_LOG=debug,regalloc2=info,cranelift_codegen::machinst=trace`.

When running tests instead of the CLI, `WASMTIME_LOG` becomes `RUST_LOG`. Most tests initialize the env-logger, but if you aren't seeing the logs, then you can put `let _ = env_logger::try_init();` at the start of the test (or ideally in a shared test helper function that a bunch of related tests already call at the beginning of their execution).

Reply:
* https://docs.wasmtime.dev/cli-logging.html?highlight=wasmtime_log#logging-in-the-wasmtime-cli
* https://docs.rs/log/latest/log/
* https://docs.rs/env_logger/latest/env_logger/